lua/opencode/ui/input_window.lua

```diff
@@ -191,6 +198,37 @@ function M.set_content(text, windows)
   local lines = type(text) == 'table' and text or vim.split(tostring(text), '\n')
 
   vim.api.nvim_buf_set_lines(windows.input_buf, 0, -1, false, lines)
+  M.add_winbar_padding(windows)
+end
+
+function M.set_current_line(text, windows)
+  windows = windows or state.windows
+  if not M.mounted(windows) then
+    return
+  end
+
+  vim.api.nvim_set_current_line(text)
+  M.add_winbar_padding(windows)
+end
+
+function M.remove_mention(mention_name, windows)
+  windows = windows or state.windows
+  if not M.mounted(windows) then
+    return
+  end
+
+  local lines = vim.api.nvim_buf_get_lines(windows.input_buf, 0, -1, false)
+  for i, line in ipairs(lines) do
+    local mention_key = config.get_key_for_function('input_window', 'mention')
+    local updated_line = line:gsub(mention_key .. mention_name, '')
```

Comment:
Need to use vim.pesc here in case `mention_name` contains - or  

```lua
    local pattern = vim.pesc(mention_key .. mention_name)
    local updated_line = line:gsub(pattern, '')
```